File: g5/g5.py
import sys

def es_valida(s: str) -> bool:
    if len(s) < 2:  # Longitud mínima es 2 ("ab")
        return False
    
    if not s.startswith('a') or not s.endswith('b'):
        return False

    # Verificar el patrón "ab" repetido
    for i in range(1, len(s) - 1, 2):
        if i + 1 >= len(s) or s[i] != 'a' or s[i + 1] != 'b':
            return False

    return True

def main():
    if len(sys.argv) != 2:
        print("ingrese un archivo")
        return
    
    nombre_archivo = sys.argv[1]
    
    try:
        with open(nombre_archivo, 'r') as f:
            for linea in f:
                cadena = linea.strip()
                resultado = "Acepta" if es_valida(cadena) else "No acepta"
                print(f"{resultado}: {cadena}")
    except FileNotFoundError:
        print(f"Error: Archivo '{nombre_archivo}' no encontrado")
    except Exception as e:
        print(f"Error: {str(e)}")

if __name__ == "__main__":
    main()

File: g5/g5.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool es_valida(const char *s) {
    int len = strlen(s);
    
    if (len < 2) return false; 
    if (s[0] != 'a' || s[len - 1] != 'b') return false; 


    for (int i = 1; i < len - 1; i += 2) {
        if (s[i] != 'a' || (i + 1 < len - 1 && s[i + 1] != 'b')) {
            return false;
        }
    }

    
    if ((len - 2) % 2 != 0) return false;

    return true;
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Uso: %s <archivo.txt>\n", argv[0]);
        return 1;
    }

    FILE *archivo = fopen(argv[1], "r");
    if (!archivo) {
        perror("Error al abrir el archivo");
        return 1;
    }

    char linea[256];
    while (fgets(linea, sizeof(linea), archivo)) {
        linea[strcspn(linea, "\n")] = '\0';
        if (es_valida(linea)) {
            printf("Acepta: %s\n", linea);
        } else {
            printf("No acepta: %s\n", linea);
        }
    }

    fclose(archivo);
    return 0;
}
